File: designpatterns/src/com/renho/behavior/state/ConcreteStateB.java
package com.renho.behavior.state;

public class ConcreteStateB extends State{

private int status;
	
	public ConcreteStateB(int status) {
		this.status = status;
	}
	
	@Override
	public void handle(Context context) {
		context.setState(new ConcreteStateC(status + 6));
		System.out.println("ÖÐÎç:" + status);
	}

	public int getStatus() {
		return status;
	}

	public void setStatus(int status) {
		this.status = status;
	}

}

File: ideaworkspace/designpatterns/src/main/scala/com/renho/behavior/state/ConcreteStateB.scala
package com.renho.behavior.state

/**
  * Created by renho on 2016/7/11.
  */
class ConcreteStateB(var status: Int) extends State {
	override def handle(context: Context): Unit = {
		println("中午:" + status)
		context.state = new ConcreteStateC(status + 6)
	}
}
